`lark.grammar.Symbol.__eq__` raises an `AssertionError` when checking if `None == Symbol(...)`

**Describe the bug**

When attempting to compare a variable set to `None` with a Symbol, Lark's implementation of `__eq__` within the `Symbol` class raises an `AssertionError`.

**To Reproduce**

```python
from lark.grammar import Symbol
a = None
b = Symbol("abc")

# this throws an error:
a == b
```

The error message:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/venv/lib/python3.8/site-packages/lark/grammar.py", line 19, in __eq__
    assert isinstance(other, Symbol), other
AssertionError: None
```

--- a/lark/grammar.py
+++ b/lark/grammar.py
@@ -16,8 +16,12 @@ class Symbol(Serialize):
         self.name = name
 
     def __eq__(self, other):
-        if not isinstance(other, Symbol):
+        if other is None:
             return False
+        if not isinstance(other, Symbol):
+            raise NotImplementedError(
+                f"Comparing a Symbol with type {type(other).__name__} is not implemented"
+            )
         return self.is_term == other.is_term and self.name == other.name
 
     def __ne__(self, other):

--- a/lark/grammar.py
+++ b/lark/grammar.py
@@ -16,12 +16,8 @@ class Symbol(Serialize):
         self.name = name
 
     def __eq__(self, other):
-        if other is None:
-            return False
         if not isinstance(other, Symbol):
-            raise NotImplementedError(
-                f"Comparing a Symbol with type {type(other).__name__} is not implemented"
-            )
+            return NotImplemented
         return self.is_term == other.is_term and self.name == other.name
 
     def __ne__(self, other):

--- a/lark/grammar.py
+++ b/lark/grammar.py
@@ -16,7 +16,8 @@ class Symbol(Serialize):
         self.name = name
 
     def __eq__(self, other):
-        assert isinstance(other, Symbol), other
+        if not isinstance(other, Symbol):
+            return False
         return self.is_term == other.is_term and self.name == other.name
 
     def __ne__(self, other):

--- a/tests/test_grammar.py
+++ b/tests/test_grammar.py
@@ -302,6 +302,14 @@ class TestGrammar(TestCase):
 
         self.assertNotEqual(a, b)
 
+        a = dict()
+
+        self.assertRaisesRegex(
+            NotImplementedError,
+            r"Comparing a Symbol with type dict is not implemented",
+            lambda: a == b,
+        )
+
 
 if __name__ == '__main__':
     main()

--- a/tests/test_grammar.py
+++ b/tests/test_grammar.py
@@ -6,7 +6,7 @@ from unittest import TestCase, main
 from lark import Lark, Token, Tree, ParseError, UnexpectedInput
 from lark.load_grammar import GrammarError, GRAMMAR_ERRORS, find_grammar_errors, list_grammar_imports
 from lark.load_grammar import FromPackageLoader
-
+from lark.grammar import Symbol
 
 class TestGrammar(TestCase):
     def setUp(self):
@@ -296,8 +296,11 @@ class TestGrammar(TestCase):
         p.parse('ab')
 
 
+    def test_symbol_eq(self):
+        a = None
+        b = Symbol("abc")
 
-
+        self.assertNotEqual(a, b)
 
 
 if __name__ == '__main__':

--- a/tests/test_grammar.py
+++ b/tests/test_grammar.py
@@ -302,14 +302,6 @@ class TestGrammar(TestCase):
 
         self.assertNotEqual(a, b)
 
-        a = dict()
-
-        self.assertRaisesRegex(
-            NotImplementedError,
-            r"Comparing a Symbol with type dict is not implemented",
-            lambda: a == b,
-        )
-
 
 if __name__ == '__main__':
     main()

